src/stravalib/model.py

```diff
@@ -374,18 +287,21 @@ def check_activity_type(cls, values: dict[str, Any]) -> dict[str, Any]:
         dict
             A dictionary with a validated activity type value assigned.
         """
-        v = values["__root__"]
-        if v not in get_args(ActivityType.__fields__["__root__"].type_):
+
+        if values not in get_args(
+            ActivityType.model_fields["root"].annotation
+        ):
             LOGGER.warning(
-                f'Unexpected activity type. Given={v}, replacing by "Workout"'
+                f'Unexpected activity type. Given={values}, replacing by "Workout"'
             )
-            values["__root__"] = "Workout"
+            values = "Workout"
         return values
 
 
+# TODO: should this type just be str (why would we accept Any here?)
```

Comment:
Same as above, what happens when `Any` is removed?

Reply:
we definitely don't need Any the way things are written now. i wasn't sure if it was there for some other purpose! i've removed it.